File: app/src/main/java/com/cielo/ordermanager/sdk/sample/adapter/PrimarySpinnerAdapter.java
package com.cielo.ordermanager.sdk.sample.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;

import com.cielo.ordermanager.sdk.R;
import cielo.orders.domain.product.PrimaryProduct;

public class PrimarySpinnerAdapter extends ArrayAdapter<PrimaryProduct> {

    private Context context;
    private List<PrimaryProduct> values;
    LayoutInflater inflater;
    PrimaryProduct tempProduct = null;

    public PrimarySpinnerAdapter(Context context, int textViewResourceId,
                                 List<PrimaryProduct> values) {
        super(context, textViewResourceId, values);
        this.context = context;
        this.values = values;

        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public int getCount() {
        return values.size();
    }

    @Override
    public PrimaryProduct getItem(int position) {
        return values.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        return getCustomView(position, convertView, parent);
    }

    @Override
    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        return getCustomView(position, convertView, parent);
    }

    public View getCustomView(int position, View convertView, ViewGroup parent) {

        View row = inflater.inflate(R.layout.spinner_item, parent, false);

        tempProduct = null;
        tempProduct = values.get(position);

        TextView label = (TextView) row.findViewById(R.id.simple_title);
        label.setText(tempProduct.getName());

        return row;
    }
}

File: app/src/main/java/com/cielo/ordermanager/sdk/adapter/PrimarySpinnerAdapter.kt
package com.cielo.ordermanager.sdk.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import cielo.orders.domain.product.PrimaryProduct
import com.cielo.ordermanager.sdk.R

class PrimarySpinnerAdapter(context: Context, textViewResourceId: Int,
                            private val values: List<PrimaryProduct>) : ArrayAdapter<PrimaryProduct>(context, textViewResourceId, values) {
    private var inflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    private var tempProduct: PrimaryProduct? = null
    override fun getCount(): Int {
        return values.size
    }

    override fun getItem(position: Int): PrimaryProduct {
        return values[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        return getCustomView(position, convertView, parent)
    }

    override fun getDropDownView(position: Int, convertView: View, parent: ViewGroup): View {
        return getCustomView(position, convertView, parent)
    }

    fun getCustomView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val row = inflater.inflate(R.layout.spinner_item, parent, false)
        tempProduct = null
        tempProduct = values[position]
        val label = row.findViewById<View>(R.id.simple_title) as TextView
        label.text = tempProduct!!.name
        return row
    }

}
